### PR TITLE
Pttb 93 backend search filter not working

### PR DESCRIPTION
**Pull request by Pranish97**

### CODE REVIEW

**Comment by Prabeg29** [comment]
@Pranish97 If there are changes due to formatting, comment out where the actual change is

File: app/Http/Controllers/ProjectController.php [hunk 1]
```diff
@@ -103,27 +100,25 @@ public function updateBillableStatus(Request $request)
...
+    public function viewAllProjects(Request $request)
     {
-        $projects = Project::paginate();
-        $count=Project::count();
-        
-        return response()->json([
-            'total' => $count,
-            'projects' => ProjectResource::collection($projects)
-        ], 200);
+        $projects = Project::latest()->paginate();
```
Comment (Prabeg29) [inline app/Http/Controllers/ProjectController.php:105]: @Pranish97 Mention the change was made here..

File: app/Http/Controllers/ProjectController.php [hunk 2]
```diff
@@ -103,27 +100,25 @@ public function updateBillableStatus(Request $request)
...
+        if ($request['search']) {
+            $projects = Project::where('project_name', 'LIKE', "%" . $request['search'] . "%")->get();
+        }
```
Comment (Pranish97) [inline app/Http/Controllers/ProjectController.php:108]: The card main changes were done between lines 106 to 108.

File: app/Http/Controllers/ProjectController.php (hunk 1 above)
Comment (Pranish97) [reply app/Http/Controllers/ProjectController.php:105]: As i was working on this card my next card was to show the latest add project to the top so I decided to finish another card too.

File: app/Http/Requests/ProjectRequest.php [hunk 3]
```diff
@@ -24,7 +24,7 @@ public function authorize()
     public function rules()
     {
         return [
-            'project_name' => 'required|max:255',
+            'project_name' => 'required |max:255||unique:projects',
```
Comment (Prabeg29) [inline app/Http/Requests/ProjectRequest.php:27]: @Pranish97 `||` ??

**Review by Prabeg29 — CHANGES_REQUESTED** [review]

File: app/Http/Requests/ProjectRequest.php (hunk 3 above)
Comment (Pranish97) [reply app/Http/Requests/ProjectRequest.php:27]: As there won't be projects with the same name.

File: app/Http/Controllers/ProjectController.php [hunk 4]
```diff
@@ -103,27 +100,25 @@ public function updateBillableStatus(Request $request)
...
-        $projects = Project::paginate();
-        $count=Project::count();
-        
-        return response()->json([
-            'total' => $count,
-            'projects' => ProjectResource::collection($projects)
-        ], 200);
+        $projects = Project::latest()->paginate();
+        if ($request['search']) {
+            $projects = Project::where('project_name', 'LIKE', "%" . $request['search'] . "%")->get();
```
Comment (Prabeg29) [inline app/Http/Controllers/ProjectController.php:107]: @Pranish97 Do paginate here aswell

**Review by Prabeg29 — CHANGES_REQUESTED** [review]

**Review by Prabeg29 — APPROVED** [review]

**Pull request merged**